File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count={}
        res=[]
        for num in nums:
            count[num] = 1 + count.get(num, 0)
        buckets=[[] for i in range(len(nums)+1)]
        for num, freq in count.items():
            buckets[freq].append(num)
        for i in range(len(buckets)-1,0,-1):
            for num in buckets[i]:
                res.append(num)
                if len(res)==k:
                    return res

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] arr = new int[n];
        int t = 1;
        int countzero = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] != 0) {
                t *= nums[i];
            } else {
                countzero++;
            }
        }
        for (int i = 0; i < n; i++) {
            if (countzero > 1) {
                arr[i] = 0;
            } else if (countzero == 1) {
                arr[i] = nums[i] == 0 ? t : 0;
            } else {
                arr[i] = t / nums[i]; 
            }
        }

        return arr;
    }
}
